[Cahya NusaBert](https://huggingface.co/cahya/NusaBert-ner-v1.3)
[Dataset Cahya NusaBert](https://huggingface.co/datasets/grit-id/id_nergrit_corpus)

In [1]:
from transformers import pipeline

ner_pipeline = pipeline("ner", model="cahya/NusaBert-ner-v1.3")

/usr/local/lib/python3.11/dist-packages/huggingface_hub/utils/_auth.py:94: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


config.json:   0%|          | 0.00/2.64k [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/644M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/19.3k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/4.70M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/695 [00:00<?, ?B/s]

Device set to use cpu


In [3]:
kalimat = "Muhammad Ikhwan Fathulloh is the founder of Microsoft"
hasil = ner_pipeline(kalimat)

print(hasil)

[{'entity': 'B-PER', 'score': np.float32(0.999977), 'index': 1, 'word': 'Muhammad', 'start': 0, 'end': 8}, {'entity': 'I-PER', 'score': np.float32(0.9994205), 'index': 2, 'word': 'ĠIkhwan', 'start': 8, 'end': 15}, {'entity': 'I-PER', 'score': np.float32(0.99916494), 'index': 3, 'word': 'ĠFath', 'start': 15, 'end': 20}, {'entity': 'I-PER', 'score': np.float32(0.9566014), 'index': 4, 'word': 'ulloh', 'start': 20, 'end': 25}, {'entity': 'B-ORG', 'score': np.float32(0.48967266), 'index': 9, 'word': 'ĠMicrosoft', 'start': 43, 'end': 53}]


[
  {'entity': 'B-PER', 'score': np.float32(0.999977), 'index': 1, 'word': 'Muhammad', 'start': 0, 'end': 8},
  {'entity': 'I-PER', 'score': np.float32(0.9994205), 'index': 2, 'word': 'ĠIkhwan', 'start': 8, 'end': 15},
  {'entity': 'I-PER', 'score': np.float32(0.99916494), 'index': 3, 'word': 'ĠFath', 'start': 15, 'end': 20},
  {'entity': 'I-PER', 'score': np.float32(0.9566014), 'index': 4, 'word': 'ulloh', 'start': 20, 'end': 25},
  {'entity': 'B-ORG', 'score': np.float32(0.48967266), 'index': 9, 'word': 'ĠMicrosoft', 'start': 43, 'end': 53}]


In [4]:
kalimat = "Elon Musk Tesla CEO visits Tokyo on August 10 2024 for new product launch"
hasil = ner_pipeline(kalimat)

print(hasil)

[{'entity': 'B-PER', 'score': np.float32(0.9991667), 'index': 1, 'word': 'E', 'start': 0, 'end': 1}, {'entity': 'I-PER', 'score': np.float32(0.88430685), 'index': 2, 'word': 'lon', 'start': 1, 'end': 4}, {'entity': 'I-PER', 'score': np.float32(0.97757643), 'index': 3, 'word': 'ĠMusk', 'start': 4, 'end': 9}, {'entity': 'B-GPE', 'score': np.float32(0.99998784), 'index': 7, 'word': 'ĠTokyo', 'start': 26, 'end': 32}, {'entity': 'B-DAT', 'score': np.float32(0.96501464), 'index': 9, 'word': 'ĠAugust', 'start': 35, 'end': 42}, {'entity': 'I-DAT', 'score': np.float32(0.9973815), 'index': 10, 'word': 'Ġ', 'start': 42, 'end': 43}, {'entity': 'I-DAT', 'score': np.float32(0.99594563), 'index': 11, 'word': '10', 'start': 43, 'end': 45}, {'entity': 'I-DAT', 'score': np.float32(0.9991875), 'index': 12, 'word': 'Ġ', 'start': 45, 'end': 46}, {'entity': 'I-DAT', 'score': np.float32(0.9994129), 'index': 13, 'word': '202', 'start': 46, 'end': 49}, {'entity': 'I-DAT', 'score': np.float32(0.99465805), 'inde

In [7]:
kalimat = "Budi membeli buku di gramedia"
hasil = ner_pipeline(kalimat)

print(hasil)

[{'entity': 'B-PER', 'score': np.float32(0.9993999), 'index': 1, 'word': 'Budi', 'start': 0, 'end': 4}, {'entity': 'B-ORG', 'score': np.float32(0.91660666), 'index': 5, 'word': 'Ġgram', 'start': 20, 'end': 25}, {'entity': 'I-LOC', 'score': np.float32(0.37086576), 'index': 6, 'word': 'edia', 'start': 25, 'end': 29}]
